.github/actions/setup-cli-deps/action.yml

```diff
@@ -24,23 +24,29 @@ runs:
       working-directory: packages/cli-kit/assets/cli-ruby
       run: bundle install
       shell: bash
+    - name: Install Node.js
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ inputs.node-version }}
     - name: Install pnpm
-      uses: pnpm/action-setup@10693b3829bf86eb2572aef5f3571dcf5ca9287d # pin@v2.2.2
+      uses: pnpm/action-setup@c3b53f6a16e57305370b4ae5a540c2077a1d50dd # pin@v2.2.4
       with:
         version: ${{ env.PNPM_VERSION }}
-    - name: Set Node.js
-      uses: actions/setup-node@master
-      with:
-        node-version: ${{ inputs.node-version }}
-        cache: 'pnpm'
-    - name: Install yarn
-      run: npm install --global yarn@${{ env.YARN_VERSION }}
-      shell: bash
-    - name: Increase yarn timeout
-      run: yarn config set network-timeout 300000
+        run_install: false
+    - name: Get pnpm store directory
+      id: pnpm-cache-path
       shell: bash
+      run: |
+        echo "STORE_PATH=$(pnpm store path)" >> $GITHUB_OUTPUT
+    - name: Setup pnpm cache
+      uses: actions/cache@v3
+      with:
+        path: ${{ steps.pnpm-cache-path.outputs.STORE_PATH }}
+        key: ${{ runner.os }}-${{ inputs.node-version }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
+        restore-keys: |
+          ${{ runner.os }}-${{ inputs.node-version }}-pnpm-store-
     - name: Install dependencies
-      run: pnpm install
+      run: pnpm install --frozen-lockfile
```

Comment:
Also this one 👏🏼

Reply:
I grepped for `pnpm install` and it's just this file and then `release.yml`. From what I could find the issue was more prominent when trying to reuse/write these caches at the same time, so I left it here. If we see weird issues when releasing I'll extract to a shared action (action-inception)